### PR TITLE
Return multiple counts

### PR DESCRIPTION
**Pull request by AndrewTorrance**
**What is the context of this PR?**

https://trello.com/c/3Bj1gYTy
There is a need to display the counts of secure message conversations in the title of each tab .
This PR enables that through a call to Secure message. 
It adds support for all_conversation_type parameter in the Thread count end point. This returns counts for open,closed,new_respondent_conversations and my conversations all in one endpoint. Its not practical to fulifil this in a single database query, a single conversation can count in multiple classes, and the selection criteria are complex. This code submits 4 queries and amalgamates the data .  

**How to review**
Run tests. New tests added for the additional parameter.

### CODE REVIEW

File: secure_message/repository/retriever.py [hunk 1]
```diff
@@ -74,6 +75,30 @@ def thread_count_by_survey(request_args, user):
...
+    def thread_count_by_survey_and_conversation_states(request_args, user):
+        """Return 4 conversation counts. They are for open, closed, my conversations and
+        new_respondent_conversation.
+        Given each of these counts uses different clauses to define them and they are not mutually exclusive,
+        they are difficult to achieve with the current db structure in a single query, hence this submits 4 db queries.
+        """
+
+        totals = {}
+
+        args = set_conversation_type_args(request_args, is_closed=False)  # is_closed shown simply for clarity
```
Comment (insacuri) [inline secure_message/repository/retriever.py]: Don't bother with the is_closed bit here.  It defaults to false in the function so this is superfluous

File: secure_message/resources/threads.py [hunk 2]
```diff
@@ -127,7 +127,24 @@ def _validate_request(request_args, user):
 
 
 class ThreadCounter(Resource):
-    """Get count of all open or closed messages"""
+    """Get count of all conversations for a specific internal user
+    Typically for a specific survey, supports filtering by case, collection exercise, business party id etc
+    Args:
```
Comment (insacuri) [inline secure_message/resources/threads.py]: I know this is picky but throughout all of our other repos we use the sphinx style of docstrings <https://www.datacamp.com/community/tutorials/docstrings-python#fifth-sub> could you convert it to that instead?

File: secure_message/repository/retriever.py (hunk 1 above)
Comment (AndrewTorrance) [reply secure_message/repository/retriever.py]: Changed,

File: secure_message/resources/threads.py (hunk 2 above)
Comment (AndrewTorrance) [reply secure_message/resources/threads.py]: Disagree . These are not specific parameters so they are not suitable for the normal :param format. What I have done is to move the comments to the get_options function. But even then they are not individual params

**Pull request merged**